### PR TITLE
Python: Update dev setup readme

### PR DESCRIPTION
**Pull request by m2web**
### Description
I made markdown updates to the DEV_SETUP.md file, updating the headings and setting the language for the code fences.

### Contribution Checklist

- [x] The code builds clean without any errors or warnings
- [ ] The PR follows the [SK Contribution Guidelines](https://github.com/microsoft/semantic-kernel/blob/main/CONTRIBUTING.md) and the [pre-submission formatting script](https://github.com/microsoft/semantic-kernel/blob/main/CONTRIBUTING.md#development-scripts) raises no violations
- [ ] All unit tests pass, and I have added new tests where possible
- [ ] I didn't break anyone :smile:

### CODE REVIEW

File: python/samples/kernel-syntax-examples/chat.py [hunk 1]
```diff
@@ -19,7 +19,7 @@
 
 api_key, org_id = sk.openai_settings_from_dot_env()
 kernel.add_chat_service(
-    "chat-gpt", sk_oai.OpenAIChatCompletion("gpt-3.5-turbo", api_key, org_id)
+    "chat-gpt", sk_oai.OpenAIChatCompletion("gpt-4-0613", api_key, org_id)
```
Comment (awharrison-28) [inline python/samples/kernel-syntax-examples/chat.py:22]: Was this change intentional? We'd like to keep this as gpt-3.5-turbo as it is more accessible than gpt-4.

**Review by awharrison-28 — CHANGES_REQUESTED** [review]

File: python/samples/kernel-syntax-examples/chat.py (hunk 1 above)
Comment (m2web) [reply python/samples/kernel-syntax-examples/chat.py:22]: Sorry, I did not mean to commit that. Please reject that change.

Comment (lemillermicrosoft) [reply python/samples/kernel-syntax-examples/chat.py:22]:
```suggestion
    "chat-gpt", sk_oai.OpenAIChatCompletion("gpt-3.5-turbo", api_key, org_id)
```

File: python/DEV_SETUP.md [hunk 2]
```diff
@@ -177,6 +187,7 @@ from pydantic import BaseModel
 class MyModel(BaseModel):
     b: B
     c: C
+```
```
Comment (lemillermicrosoft) [inline python/DEV_SETUP.md:190]:
```suggestion
```

**Review by lemillermicrosoft — APPROVED** [review]

**Review by awharrison-28 — APPROVED** [review]
